.github/workflows/macos-cmake.yml

```diff
@@ -0,0 +1,149 @@
+name: macOS CMake release builds
+
+on:
+  push:
+    paths-ignore:
+      - '.clang-format'
+      - '.mdl-styles'
+      - '*.md'
+      - 'docs/**'
+      - 'licenses/**'
+      - 'website/**'
+
+  pull_request:
+    paths-ignore:
+      - '.clang-format'
+      - '.mdl-styles'
+      - '*.md'
+      - 'docs/**'
+      - 'licenses/**'
+      - 'website/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_macos_release:
+    name: Release build (${{ matrix.runner.arch }})
+    runs-on: ${{ matrix.runner.host }}
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    env:
+      MACOSX_DEPLOYMENT_TARGET: ${{ matrix.runner.minimum_deployment }}
+      VCPKG_BINARY_SOURCES: "clear;x-gha,readwrite"
+      VCPKG_ROOT: ${{ github.workspace }}/vcpkg
+
+    strategy:
+      matrix:
+        runner:
+          - host: macos-15
+            arch: x86_64
+            minimum_deployment: '10.15'
+
+          - host: macos-15
+            arch: arm64
+            minimum_deployment: '11.0'
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: false
+
+      - name: Export GitHub Actions cache environment variables
+        uses: actions/github-script@v7
+        with:
+          script: |
+            core.exportVariable('ACTIONS_CACHE_URL', process.env.ACTIONS_CACHE_URL || '');
+            core.exportVariable('ACTIONS_RUNTIME_TOKEN', process.env.ACTIONS_RUNTIME_TOKEN || '');
+  
+      - name: Setup CMake
+        uses: lukka/get-cmake@v3.31.5
+
+      - name: Clone vcpkg
+        run: git clone https://github.com/microsoft/vcpkg.git $VCPKG_ROOT
+  
+      - name: Bootstrap vcpkg
+        run: |
+          cd $VCPKG_ROOT
+          ./bootstrap-vcpkg.sh         
+            
+      - name: Log environment
+        run:  arch -arch=${{ matrix.runner.arch }} ./scripts/log-env.sh
+
+      - name: Inject version string
+        run: |
+          set -x
+          VERSION=$(./scripts/get-version.sh version-and-hash)
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+
+      - name: Setup and build release
+        run: |
+          set -x
+          cmake --preset release-macos-${{ matrix.runner.arch }}          
+          cmake --build --preset release-macos-${{ matrix.runner.arch }}
+
+      - name: Upload binary
+        uses: actions/upload-artifact@v4
+        with:
+          name: dosbox-${{ matrix.runner.arch }}
+          path: build/release-macos-${{ matrix.runner.arch }}/Release/dosbox
+          overwrite: true
```

Comment:
Wouldn't it be better to make the git hash part of the name? Just cleaner and less potentialy surprises. Not even sure if concurrent runs are a thing, but that would cause problems with such "global" state.

Reply:
Possibly, but this was just a copy-paste from the main macos.yml, I didn't try to "optimize" anything - yet!

Reply:
Fair enough 😄 I was thinking of code optimisations when you mentioned "optimisations" in the above description (e.g. passing compiler flags), but makes sense now. We can do all this later, sure thing.